framework/Exceptions/TException.php

```diff
@@ -67,18 +85,23 @@ public function __construct($errorMessage)
 	 */
 	protected function translateErrorMessage($key)
 	{
-		$msgFile = $this->getErrorMessageFile();
+		$msgFiles = TException::$_messagefiles;
+		$msgFiles[] = $this->getErrorMessageFile();
+		$value = $key;
 
 		// Cache messages
-		if (!isset(self::$_messageCache[$msgFile])) {
-			if (($entries = @file($msgFile)) !== false) {
-				foreach ($entries as $entry) {
-					[$code, $message] = array_merge(explode('=', $entry, 2), ['']);
-					self::$_messageCache[$msgFile][trim($code)] = trim($message);
+		foreach ($msgFiles as $msgFile) {
+			if (!isset(self::$_messageCache[$msgFile])) {
+				if (($entries = @file($msgFile)) !== false) {
+					foreach ($entries as $entry) {
+						[$code, $message] = array_merge(explode('=', $entry, 2), ['']);
+						self::$_messageCache[$msgFile][trim($code)] = trim($message);
+					}
+					$value = self::$_messageCache[$msgFile][$key] ?? $value;
 				}
 			}
 		}
-		return self::$_messageCache[$msgFile][$key] ?? $key;
+		return $value;
```

Comment:
This looks like an error, if the $_messageCache is already populated $value is never set

Reply:
Line 90 has $value = $key;
and line 100 has $value = self::$_messageCache[$msgFile][$key] ?? $value;

Looping through the possible multiple files is an important aspect of the new method

Reply:
Line 100 get executed only if `!isset(self::$_messageCache[$msgFile])`
So if the file has already been cached before, it's never executed (or maybe I'm overlooking this?)

Reply:
my thinking may be off with this... let me see...

Reply:
I'm glad we are going over these.  where ever my brain was for that error is beyond me.  lol.  I have the correction here.

Reply:
corrected